### PR TITLE
Add support for QQP dataset with unit tests

### PR DESCRIPTION
**Pull request by VirgileHlav**
## Summary
- Added support for QQP dataset
- Added mocked unit tests for QQP dataset 

## Test
- ` pytest test/datasets/test_qqp.py`

## Context
See #1710

### CODE REVIEW

**Review by parmeet — APPROVED** [review]
LGTM!

**Review by Nayef211 — APPROVED** [review]
LGTM. Can we also make sure to add the sharding filter to the dataset as @parmeet mentioned here https://github.com/pytorch/text/issues/1710#issuecomment-1128958655

File: test/datasets/test_qqp.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    root_dir: directory to the mocked dataset
+    """
+    base_dir = os.path.join(root_dir, "QQP")
+    os.makedirs(base_dir, exist_ok=True)
+
+    seed = 1
+    file_name = "quora_duplicate_questions.tsv"
+    txt_file = os.path.join(base_dir, file_name)
+    mocked_data = []
+    print(txt_file)
```
Comment (Nayef211) [inline test/datasets/test_qqp.py:21]: Can we remove the print statement here

File: test/datasets/test_qqp.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+class TestQQP(TempDirMixin, TorchtextTestCase):
+    root_dir = None
+    samples = []
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.root_dir = cls.get_base_temp_dir()
+        print(cls.root_dir)
```
Comment (Nayef211) [inline test/datasets/test_qqp.py:45]: Remove print

Comment (VirgileHlav) [reply test/datasets/test_qqp.py:45]: Solved with #1734

File: test/datasets/test_qqp.py (hunk 1 above)
Comment (VirgileHlav) [reply test/datasets/test_qqp.py:21]: Solved with #1734

**Pull request merged**